build-scripts/build_xccdf.py

```diff
@@ -143,12 +115,23 @@ def main():
     loader.load_benchmark(benchmark_root)
 
     loader.add_fixes_to_rules()
+    xccdftree = loader.export_benchmark_to_xml()
 
-    if args.thin_ds_components_dir != "off":
-        link_benchmark_per_profile(loader, args)
+    checks = xccdftree.findall(".//{%s}check" % ssg.constants.XCCDF12_NS)
 
-    xccdftree = loader.export_benchmark_to_xml()
-    link_benchmark(loader, xccdftree, args)
+    oval_linker = link_oval(xccdftree, checks, args.oval, args.build_ovals_dir)
+
+    ocil = loader.export_ocil_to_xml()
+    link_ocil(xccdftree, checks, args.ocil, ocil)
+
+    ssg.xml.ElementTree.ElementTree(xccdftree).write(args.xccdf)
```

Comment:
When I think of it, I realize that in case of building thin data streams we don't want to store this file and moreover we don't need to serialize it. But I can see that most of the code depends on the `xccdftree`. Therefore, I'm afraid that it would be difficult to rework this now. But I think it is a nice idea for future tickets. WDYT?

Reply:
Yes, this xccdf file is created because it is used in the next steps of the build. Especially in the formatting steps. These steps might be removed when python2 is not supported.